src/node/internal/internal_buffer.ts

```diff
@@ -2294,6 +2294,18 @@ export function isUtf8(value: ArrayBufferView) {
   return bufferUtil.isUtf8(value);
 }
 
+export function transcode(source: ArrayBufferView, fromEncoding: string, toEncoding: string) {
+  const normalizedFromEncoding = normalizeEncoding(fromEncoding);
+  if (!Buffer.isEncoding(normalizedFromEncoding)) {
+    throw new ERR_UNKNOWN_ENCODING(fromEncoding);
+  }
+  const normalizedToEncoding = normalizeEncoding(toEncoding);
+  if (!Buffer.isEncoding(normalizedToEncoding)) {
+    throw new ERR_UNKNOWN_ENCODING(toEncoding);
+  }
+  return bufferUtil.transcode(source, fromEncoding, toEncoding);
```

Comment:
```suggestion
  return bufferUtil.transcode(source, normalizedFromEncoding, normalizedToEncoding);
```

?

Reply:
Although, I do wonder if we should normalize these to int values matching the internal enum rather than passing these through as strings. Passing them down as strings means we will always be additionally copying the values and performing an unnecessary allocation and utf8 write in order to get those as kj::String internally. If we know they are valid encodings at this point, using the enum would be more efficient.

Reply:
Yup reducing the string serialization cost would be beneficial. Is there an easy way to share ENUM values with JS without duplication?

Reply:
Look for `JSG_STATIC_CONSTANT` or `JSG_STATIC_CONSTANT_NAMED` ... I think that's what you might want here.

Reply:
I added a comment for this. I'll follow-up after this pull-request lands.